tensorboard/plugins/text/tf_text_dashboard/tf-text-dashboard.html

```diff
@@ -29,73 +28,120 @@
+                <tf-text-loader
+                  tag="[[item.tag]]"
+                  run="[[item.run]]"
+                  request-manager="[[_requestManager]]"
+                ></tf-text-loader>
+              </template>
+            </div>
+          </tf-collapsable-pane>
+        </template>
       </div>
     </tf-dashboard-layout>
     <style include="dashboard-style"></style>
     <style>
-      tf-panes-helper {
-        --card-width: 100%;
-        --card-height: auto;
-        --card-expanded-width: 100%;
-        --card-expanded-height: 1000px;
-        --card-padding: 0 5px 5px 5px;
-        --show-expand-button: none;
+      .no-data-warning {
+        max-width: 540px;
+        margin: 80px auto 0 auto;
       }
     </style>
   </template>
   <script>
-    import {DashboardBehavior} from "../tf-dashboard-common/dashboard-behavior.js";
-    import {ReloadBehavior} from "../tf-dashboard-common/reload-behavior.js";
-    import {BackendBehavior} from "../tf-backend/behavior.js";
+    import {RequestManager} from '../tf-backend/requestManager.js';
+    import {getTags} from '../tf-backend/backend.js';
+    import {getRouter} from '../tf-backend/router.js';
+    import {categorizeRunTagCombinations} from '../tf-categorization-utils/categorizationUtils.js';
 
     Polymer({
       is: "tf-text-dashboard",
       properties: {
-        backend: Object,
-        dataType: {
-          type: String,
-          value: "text"
+        _selectedRuns: Array,
+        _runToTag: Object,  // map<run: string, tags: string[]>
+        _dataNotFound: Boolean,
+
+        _categories: {
+          type: Array,
+          computed:
+            '_makeCategories(_runToTag, _selectedRuns, _tagGroupRegexes)',
+        },
+
+        _requestManager: {
+          type: Object,
+          value: () => new RequestManager(),
         },
       },
-      behaviors: [
-        DashboardBehavior("text"),
-        ReloadBehavior("tf-chart-scaffold"),
-        BackendBehavior,
-      ],
-      attached: function() {
-        this.async(function() {
-          this.fire("rendered");
+      ready() {
+        this.reload();
+      },
+      reload() {
+        this._fetchTags().then(() => {
+          this._reloadTexts();
         });
       },
+      _fetchTags() {
+        const url = getRouter().pluginRoute('text', '/tags');
+        return this._requestManager.request(url).then(runToTag => {
+          if (_.isEqual(runToTag, this._runToTag)) {
+            // No need to update anything if there are no changes.
+            return;
+          }
+          const tags = getTags(runToTag);
+          this.set('_dataNotFound', tags.length === 0);
+          this.set('_runToTag', runToTag);
+        });
+      },
+      _reloadTexts() {
+        this.querySelectorAll('tf-text-loader').forEach(textLoader => {
+          textLoader.reload();
+        });
+      },
+      _makeCategories(runToTag, selectedRuns, tagGroupRegexes) {
```

Comment:
I wish polymer didn't make us redefine functions on the object in order to use them... (No change requested)